Fix failure of the c2s listener to work when
starttls_required is specified along with zlib in the
listener's configuration.

This issue is tracked upstream as EJAB-1118.

This changeset was introduced in revision 2789,
branch http://svn.process-one.net/ejabberd/branches/ejabberd-2.1.x

The first stable version containing the fix is 2.1.1.

--- a/src/ejabberd_c2s.erl	(revision 2788)
+++ b/src/ejabberd_c2s.erl	(revision 2789)
@@ -187,7 +187,6 @@
 			(_) -> false
 		     end, Opts),
     TLSOpts = [verify_none | TLSOpts1],
-    Zlib = lists:member(zlib, Opts) andalso (not StartTLSRequired),
     IP = peerip(SockMod, Socket),
     %% Check if IP is blacklisted:
     case is_ip_blacklisted(IP) of
